# Generated by Django 3.2.5 on 2021-07-28 11:53

from django.conf import settings
import django.core.validators
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        ('products', '__first__'),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name='Organization',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('province', models.CharField(max_length=100)),
                ('name', models.CharField(max_length=150)),
                ('organization_phone', models.CharField(max_length=11, validators=[django.core.validators.RegexValidator(message='phone number invalid', regex='^0[0-9]{2,}[0-9]{7,}$')])),
                ('employees_number', models.CharField(max_length=10000)),
                ('owner', models.CharField(max_length=150)),
                ('email', models.EmailField(max_length=254)),
                ('owner_phone', models.CharField(max_length=11, validators=[django.core.validators.RegexValidator(message='phone number invalid', regex='^0[0-9]{2,}[0-9]{7,}$')])),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('created_by', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
                ('organization_product', models.ManyToManyField(to='products.FollowUpProduct')),
            ],
        ),
        migrations.AddConstraint(
            model_name='organization',
            constraint=models.UniqueConstraint(fields=('name', 'created_by'), name='UniqueOrgan'),
        ),
    ]
